presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatSelectiveStreamReader.java

```diff
@@ -244,9 +244,8 @@ public int read(int offset, int[] positions, int positionCount)
         return outputPositionCount;
     }
 
-    private void readAllEmpty(int[] positions, int positionCount)
+    private void readAllEmpty(int positionCount)
     {
-        outputPositions = positions;
```

Comment:
If my understanding is right, this is the fix. Just removing this line, would have fixed the issue.